internal/fluentbit/config/builder/config_builder.go

```diff
@@ -1,129 +1,113 @@
-		return "", err
-	}
+func (f *FluentBitConfig) addEnvConfigSecret(envConfigSecret map[string][]byte) {
+	maps.Copy(f.EnvConfigSecret, envConfigSecret)
+}
 
-	err = validateInput(pipeline)
-	if err != nil {
-		return "", err
-	}
+func (f *FluentBitConfig) addTLSConfigSecret(tlsConfigSecret map[string][]byte) {
+	maps.Copy(f.TLSConfigSecret, tlsConfigSecret)
+}
 
-	err = validateCustomSections(pipeline)
-	if err != nil {
-		return "", err
+func NewFluentBitConfigBuilder(client client.Reader) *ConfigBuilder {
+	return &ConfigBuilder{
+		Reader: client,
+		BuilderConfig: BuilderConfig{
+			PipelineDefaults: PipelineDefaults{
+				InputTag:          defaultInputTag,
+				MemoryBufferLimit: defaultMemoryBufferLimit,
+				StorageType:       defaultStorageType,
+				FsBufferLimit:     defaultFsBufferLimit,
+			},
+		},
 	}
-
-	includePath := createIncludePath(pipeline)
-	excludePath := createExcludePath(pipeline, config.CollectAgentLogs)
-
-	var sb strings.Builder
-
-	sb.WriteString(createInputSection(pipeline, includePath, excludePath))
-	// skip if the filter is a multiline filter, multiline filter should be first filter in the pipeline filter chain
-	// see for more details https://docs.fluentbit.io/manual/pipeline/filters/multiline-stacktrace
-	sb.WriteString(createCustomFilters(pipeline, multilineFilter))
-	sb.WriteString(createRecordModifierFilter(pipeline))
-	sb.WriteString(createKubernetesFilter(pipeline))
-	sb.WriteString(createCustomFilters(pipeline, nonMultilineFilter))
-	sb.WriteString(createLuaDedotFilter(pipeline))
-	sb.WriteString(createOutputSection(pipeline, config.PipelineDefaults))
-
-	return sb.String(), nil
 }
 
-func createRecordModifierFilter(pipeline *telemetryv1alpha1.LogPipeline) string {
-	return NewFilterSectionBuilder().
-		AddConfigParam("name", "record_modifier").
-		AddConfigParam("match", fmt.Sprintf("%s.*", pipeline.Name)).
-		AddConfigParam("record", "cluster_identifier ${KUBERNETES_SERVICE_HOST}").
-		Build()
+type ConfigBuilder struct {
+	client.Reader
+	BuilderConfig
 }
 
-func createLuaDedotFilter(logPipeline *telemetryv1alpha1.LogPipeline) string {
-	output := logPipeline.Spec.Output
-	if !logpipelineutils.IsHTTPDefined(&output) || !output.HTTP.Dedot {
-		return ""
+func (b *ConfigBuilder) Build(ctx context.Context, allPipelines []telemetryv1alpha1.LogPipeline) (*FluentBitConfig, error) {
+	config := FluentBitConfig{
+		SectionsConfig:  make(map[string]string),
+		FilesConfig:     make(map[string]string),
+		EnvConfigSecret: make(map[string][]byte),
+		TLSConfigSecret: make(map[string][]byte),
 	}
 
-	return NewFilterSectionBuilder().
-		AddConfigParam("name", "lua").
-		AddConfigParam("match", fmt.Sprintf("%s.*", logPipeline.Name)).
-		AddConfigParam("script", "/fluent-bit/scripts/filter-script.lua").
-		AddConfigParam("call", "kubernetes_map_keys").
-		Build()
-}
+	for _, pipeline := range allPipelines {
+		if !isLogPipelineReconcilable(allPipelines, &pipeline) || !pipeline.DeletionTimestamp.IsZero() {
```

Comment:
The check for the deletion timestamp can be moved to the `isLogPipelineReconcilable` func